tests/Database/RelationsTest.php

```diff
@@ -0,0 +1,184 @@
+    }
+
+    public function testTablesExist()
+    {
+        $this->assertTrue($this->db->schema()->hasTable('posts'));
+        $this->assertTrue($this->db->schema()->hasTable('terms'));
+        $this->assertTrue($this->db->schema()->hasTable('posts_terms'));
+    }
+
+    public function testTablesProperlySeeded()
+    {
+        $this->assertEquals(1, Post::count());
+        $this->assertEquals(6, Term::count());
+        $this->assertEquals(2, Term::where('type', 'tag')->count());
+        $this->assertEquals(4, Term::where('type', 'category')->count());
+    }
+
+    public function testBelongsToManyCount()
+    {
+        $post = Post::first();
+        $this->assertEquals(2, $post->tags->count());
+        $this->assertEquals(2, $post->categories->count());
+    }
+
+    public function testBelongsToManySyncAll()
+    {
+        $post = Post::first();
+
+        $id = $post->categories()->first()->id;
+        $post->categories()->sync([$id]);
+        $this->assertEquals(1, $post->categories->count());
+        $this->assertEquals($id, $post->categories()->first()->id);
+
+        $id = $post->tags()->first()->id;
+        $post->tags()->sync([$id]);
+        $this->assertEquals(1, $post->tags->count());
+        $this->assertEquals($id, $post->tags()->first()->id);
+    }
+
+    public function testBelongsToManySyncTags()
+    {
+        $post = Post::first();
+
+        $post->categories()->detach();
+        $this->assertEquals(0, $post->categories->count());
+
+        $id = $post->tags()->first()->id;
+        $post->tags()->sync([$id]);
+        $this->assertEquals(1, $post->tags->count());
+        $this->assertEquals($id, $post->tags()->first()->id);
+    }
+
+    public function testBelongsToManySyncCategories()
+    {
+        $post = Post::first();
+
+        $id = $post->categories()->first()->id;
+        $post->categories()->sync([$id]);
+        $this->assertEquals(1, $post->categories->count());
+        $this->assertEquals($id, $post->categories()->first()->id);
+
+        $post->tags()->detach();
+        $this->assertEquals(0, $post->tags->count());
+    }
+
+    public function testBelongsToManyDetach()
+    {
+        $post = Post::first();
+
+        $post->categories()->detach();
+        $this->assertEquals(0, $post->categories->count());
+
+        $post->tags()->detach();
+        $this->assertEquals(0, $post->tags->count());
+    }
+
+    public function testBelongsToManySyncMultipleCategories()
+    {
+        $post = Post::first();
+
+        $category_ids = Term::where('type', 'category')->lists('id');
+        $this->assertEquals(4, count($category_ids));
+
+        $post->categories()->sync($category_ids);
+        $this->assertEquals(4, $post->categories()->count());
+    }
+
+    public function testBelongsToManyDetachOneCategory()
+    {
+        $post = Post::first();
+
+        $post->categories()->detach([6]);
```

Comment:
@mjauvin I would recommend not using a static ID for tests - it makes them too susceptible to failure should any further changes to this particular test suite be done. Also, do you know if the tables are reset after each test? An ID of `6` seems to be a bit weird if this were a fresh DB.

Reply:
Hey @bennothommo, you're right but until I find how to flush the cache between queries, it's hard to do otherwise.

The number 6 is because I add 6 records to the Term model in the seedTables() method.

Thanks!

Reply:
I used this in the mean time, at least it's not using a static id.

Reply:
@LukeTowers do you know how to flush the query cache after a db query?

Reply:
@mjauvin https://octobercms.com/docs/database/query#in-memory-caching

Reply:
Thanks @LukeTowers, exactly what I was looking for!